File: back-end/src/main/java/es/ull/project/application/service/workflow/WorkFlowGetService.java
package es.ull.project.application.service.workflow;

import es.ull.project.application.repository.WorkFlowRepository;
import es.ull.project.application.usecase.workflow.WorkFlowGetUseCase;
import es.ull.project.domain.entity.WorkFlow;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

import org.springframework.beans.factory.annotation.Autowired;

public class WorkFlowGetService implements WorkFlowGetUseCase {

    @Autowired
    private WorkFlowRepository repository;

    @Override
    public Optional<WorkFlow> getWorkFlow(UUID workflow) {
        return Optional.ofNullable(repository.fetchById(workflow));
    }

    @Override
    public List<WorkFlow> getAllWorkFlows() {
        return this.repository.fetchAll();
    }
}

File: front-end/src/application/service/workflow/workflow-get-service.ts
import type { WorkFlowGetUseCase } from '../../usecase/workflow/workflow-get-use-case';
import { WorkFlow } from '../../../domain/entity/workflow';
import type { WorkFlowRepository } from '../../repository/workflow-repository';
import { Either } from '@ull-tfg/ull-tfg-typescript';
import type { DataError, UllUUID } from '@ull-tfg/ull-tfg-typescript';

export class FlowGetService implements WorkFlowGetUseCase {

  constructor(private readonly flowRepository: WorkFlowRepository) {}

  async getWorkFlow(flowId: UllUUID): Promise<Either<DataError, WorkFlow>> {
    return this.flowRepository.getWorkFlow(flowId);
  }

  async getAllWorkFlows(): Promise<Either<DataError, WorkFlow[]>> {
    return this.flowRepository.getWorkFlows();
  }
}
